### DISCUSSION TITLE
Make compatible with `Do Not Disturb`

### DISCUSSION THREAD

**Discussion by BobSquarePants** [pilot51/voicenotify#138] [Ideas]
Hi,

Natively in Android exist the **Do Not Disturb** feature. Within you can make some exception, peoples, apps, etc..

So I add some contact in the exceptions. Sadly [VoiceNotify](https://github.com/pilot51/voicenotify) do not handle this.

In **VoiceNotify\Devices States** if we check **Silent/Vibrate** it will trigger for all (outside the scope of DND[^1] ) and if we uncheck it the DND[^1] exceptions are not working... Any workaround for now ? or maybe this can be an extra feature ?

Thanks

[^1]:Do Not Disturb

**Comment by pilot51** [comment]
Per-app settings is currently in the v1.4.0 beta. That would allow you to set the Silent/Vibrate option per app.

Though yes, VN is totally unaware of the DND exceptions. I'll have to look into feasibility, but perhaps I could add another option for it to follow the exceptions.

**Comment by rakolambski** [comment]
In the spirit of "never too much customization" (and if it's easy), I could see yet another checkbox option to allow per-app VN option to follow either the (current style) time limited silence silence OR follow DND rules.  Best of both worlds. When I'm in a meeting, VN doesn't interrupt, but I'm awake and can see an alert even in DND mode, so the main purpose would be to hear something important in the middle of the night. 

Having a low/med/high value with different rules for each could be a workable alternative (especially if matching rules could dynamically adjust the priority of an event), but now I'm just daydreaming. :⁠-⁠)
